File: Task 1/C++/Components/Product/Storage.cpp
#include "Storage.h"

Storage::Storage() :Furniture(){
    capacity = 0;
}

Storage::Storage(std::string name, float cost, std::string supplier, float discount_amount, int packing_units, float cost_per_unit, std::string material, int capacity)
      : Furniture(name, cost, supplier, discount_amount, packing_units, cost_per_unit, material){
    this->capacity = capacity;
}

Storage::Storage(const Storage& other) : Furniture(other) {
    this->capacity = other.capacity;
}

std::string Storage::display_product_info(){
    return Furniture::display_product_info() +
           "\nStorage Capacity: " + std::to_string(capacity) + " units";
}

File: Task 1/Java/logistics/src/main/java/com/logistics/Components/Product/Storage.java
package com.logistics.Components.Product;

public class Storage extends Furniture {

    private int capacity;

    public Storage() {
        super();
        capacity = 0;
    }

    public Storage(String name, float cost, String supplier, float discountAmount, int packingUnits, float costPerUnit, String material, int capacity) {
        super(name, cost, supplier, discountAmount, packingUnits, costPerUnit, material);
        this.capacity = capacity;
    }

    // Copy Constructor
    public Storage(Storage other) {
        super(other);
        this.capacity = other.capacity;
    }

    // Clone method
    @Override
    public Storage clone() {
        return new Storage(this);
    }

    // Getters
    public int getCapacity() { return capacity; }

    // Setters
    public void setCapacity(int capacity) { this.capacity = capacity; }

    @Override
    public String display_product_info() {
        return super.display_product_info() + 
                "\nCapacity: " + capacity;
    }
}
